packages/babel-helpers/src/helpers/iterableToArrayLimit.ts

```diff
@@ -0,0 +1,49 @@
+/* @minVersion 7.0.0-beta.0 */
+
+export default function _iterableToArrayLimit<T>(arr: Iterable<T>, i: number) {
+  // this is an expanded form of \`for...of\` that properly supports abrupt completions of
+  // iterators etc.
+
+  var iterator: Iterator<T> =
+    arr == null
+      ? null
+      : (typeof Symbol !== "undefined" && arr[Symbol.iterator]) ||
+        (arr as any)["@@iterator"];
+  if (iterator == null) return;
+
+  var _arr: T[] = [];
+  var iteratorNormalCompletion = true;
+  var didIteratorError = false;
+  var step, iteratorError, next, _return;
+  try {
+    next = (iterator = (iterator as unknown as Function).call(arr)).next;
+    if (i === 0) {
+      if (Object(iterator) !== iterator) return;
+      iteratorNormalCompletion = false;
+    } else {
+      for (
+        ;
+        !(iteratorNormalCompletion = (step = next.call(iterator)).done);
+        iteratorNormalCompletion = true
+      ) {
+        _arr.push(step.value);
+        if (_arr.length === i) break;
+      }
```

Comment:
```suggestion
      while (!(iteratorNormalCompletion = (step = next.call(iterator)).done)) {
        _arr.push(step.value);
        if (_arr.length === i) break;
        iteratorNormalCompletion = true;
      }
```

Given that we are running a minifier on this code we can now make it a bit more readable -- the minifier will convert if to `for(;...)` anyway.

Reply:
Or maybe we can out-smart the minifier, does it generate smaller code if we do this?
```suggestion
      while (
        i < _arr.length &&
        !(iteratorNormalCompletion = true, iteratorNormalCompletion = (step = next.call(iterator)).done)
      ) {
        _arr.push(step.value);
      }
```
and removing `= true` from where `var iteratorNormalCompletion` is defined.